endpoints.go

```diff
@@ -33,3 +42,312 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
 	healthInfo := models.Health{Database: dbStatus, RabbitMQ: viper.GetString("dummy_health_rabbitmq")}
 	json.NewEncoder(w).Encode(healthInfo)
 }
+
+//Find survey by reference, short name, long name, or any combination of the three
+func getSurvey(w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    queryParams := r.URL.Query()
+    if len(queryParams) == 0 {
+        w.WriteHeader(http.StatusBadRequest)
+        errorString := models.Error{
+            Error: "No query parameters provided for search",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    var sb strings.Builder
+    sb.WriteString(" WHERE 1=1")
+    for k := range queryParams {
+        switch k {
+        case "surveyRef":
+            sb.WriteString(" AND survey_ref='")
+            sb.WriteString(queryParams.Get("surveyRef"))
+            sb.WriteString("'")
+        case "shortName":
+            sb.WriteString(" AND short_name='")
+            sb.WriteString(queryParams.Get("shortName"))
+            sb.WriteString("'")
+        case "longName":
+            sb.WriteString(" AND long_name='")
+            sb.WriteString(queryParams.Get("longName"))
+            sb.WriteString("'")
+        default:
+            w.WriteHeader(http.StatusBadRequest)
+            errorString := models.Error{
+                Error: "Invalid query parameter " + k,
+            }
+            json.NewEncoder(w).Encode(errorString)
+            return
+        }
+    }
+
+    queryString := "SELECT survey_ref, short_name, long_name, legal_basis, survey_mode FROM surveyv2.survey" + sb.String()
+
+    rows, err := db.Query(queryString)
+
+    if err != nil {
+        http.Error(w, "get survey query failed", http.StatusInternalServerError)
+        return
+    }
+
+    var listOfSurveys = []models.Survey{}
+
+    for rows.Next(){
+        survey := models.Survey{}
+
+        rows.Scan(
+            &survey.Reference,
+            &survey.ShortName,
+            &survey.LongName,
+            &survey.LegalBasis,
+            &survey.SurveyMode,
+        )
+
+        listOfSurveys = append(listOfSurveys, survey)
+    }
+
+    if len(listOfSurveys) == 0 {
+        re := models.NewRESTError("404", "Survey not found")
```

Comment:
I wouldn't put the status code in an error response - you're already putting it in the header.